flag-app/flag-app-project/src/components/Filter.tsx

```diff
@@ -1,24 +1,35 @@
-import { useDarkMode } from "../hooks";
+import { useCountries, useDarkMode } from "../hooks";
 import { RiArrowDropDownLine } from "react-icons/ri";
 
 export const Filter = () => {
   const { mode } = useDarkMode();
+  const { filterValue, handleFilter } = useCountries();
+
+  const baseClassName =
+    "pl-5 pt-5 pb-5 bg-white w-56 rounded-lg shadow-sm appearance-none";
+  const darkModeClassName =
+    mode === "dark" ? "bg-darkModeBlack shadow-white" : "bg-white";
+
+  const regions = [
+    { value: "", label: "Filter by Region" },
```

Comment:
So this regions.value is exactly what should go into filterValue, as in my first comment. Labels should only be used for "presentation", and `values` are what you use in functions and other logic